src/closure/createClosure.ts

```diff
@@ -561,6 +556,13 @@ async function analyzeFunctionInfoAsync(
     // i.e. the inner call to "f();" will actually call the *outer* __f function, and not
     // itself.
     if (functionDeclarationName !== undefined) {
+      const funcEntry = context.cache.get(func);
+      if (!funcEntry) {
+        throw new Error(
+          "Entry for this this function was not created by caller"
+        );
+      }
+
```

Comment:
Why does moving this here change things?

Reply:
They process all functions beforehand and cache the results. We are injecting a function that was not processed.

The cached entry is only used when the function is named, so I moved it to fail at the last moment. It may still fail later, but this unblocks us.